libsrc/vic20/tgi/vic20-hi.s

```diff
@@ -0,0 +1,1054 @@
+;
+; Graphics driver for a 160x192x2 mode on the VIC-20.
+;
+; Based on C64 TGI driver
+;
+; 2018-03-11, Sven Michael Klose
+;
+
+        .include        "zeropage.inc"
+
+        .include        "tgi-kernel.inc"
+        .include        "tgi-error.inc"
+
+        .macpack        generic
+        .macpack        module
+
+
+; ------------------------------------------------------------------------
+; Constants
+
+COLS := 20
+ROWS := 12
+XRES := COLS * 8
+YRES := ROWS * 16
+
+TGI_IOCTL_VIC20_SET_PATTERN     :=  $01
+
+; ------------------------------------------------------------------------
+; Header. Includes jump table and constants.
+
+        module_header   _vic20_hi_tgi
+
+; First part of the header is a structure that has a magic and defines the
+; capabilities of the driver
+
+        .byte   $74, $67, $69           ; "tgi"
+        .byte   TGI_API_VERSION         ; TGI API version number
+        .addr   $0000                   ; Library reference
+        .word   XRES                    ; X resolution
+        .word   YRES                    ; Y resolution
+        .byte   2                       ; Number of drawing colors
+        .byte   1                       ; Number of screens available
+        .byte   8                       ; System font X size
+        .byte   8                       ; System font Y size
+        .word   $00D4                   ; Aspect ratio (based on 4/3 display)
```

Comment:
I'm pretty sure this value is incorrect. Please adjust it (or elaborate why you consider it correct).